test/azure/resourceid_test.go

```diff
@@ -0,0 +1,96 @@
+// +build azure
+
+// NOTE: We use build tags to differentiate azure testing because we currently do not have azure access setup for
+// CircleCI.
+
+package test
+
+import (
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/azure"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestGetNameFromResourceID(t *testing.T) {
+	// set slice variables
+	sliceSource := "this/is/a/long/slash/separated/string/ResourceID"
+	sliceResult := "ResourceID"
+	sliceNotFound := "noresourcepresent"
+
+	// verify success
+	resultSuccess := azure.GetNameFromResourceID(sliceSource)
+	assert.Equal(t, sliceResult, resultSuccess)
+
+	// verify error when seperator not found
+	resultBadSeperator := azure.GetNameFromResourceID(sliceNotFound)
+	assert.Equal(t, "", resultBadSeperator)
+}
+
+func TestGetSliceLastValue(t *testing.T) {
```

Comment:
Please make use of [table-driven tests](https://github.com/golang/go/wiki/TableDrivenTests).

Reply:
Done, please see: modules/collections/stringslicevalue_test.go